spring-security-oauth2/src/main/java/org/springframework/security/oauth2/common/OAuth2AccessTokenJackson2Deserializer.java

```diff
@@ -92,7 +92,7 @@ else if (OAuth2AccessToken.SCOPE.equals(name)) {
 
 		DefaultOAuth2AccessToken accessToken = new DefaultOAuth2AccessToken(tokenValue);
 		accessToken.setTokenType(tokenType);
-		if (expiresIn != null) {
+		if (expiresIn != null && expiresIn > 0) {
```

Comment:
Can you please add a test

Reply:
Hi.
Sure. I'l try to do this in next three days.

Reply:
@jgrandja 
Cant figure out how to edit this PR (original fork rep was removed)
So a make another one with this fix and test
#1862